### PR TITLE
chore: Make fake mods

### PR DESCRIPTION
**Pull request by bradbeam**
Signed-off-by: Brad Beam <brad.beam@talos-systems.com>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -17,9 +17,12 @@ RUN make -j $(($(nproc) / 2))
 COPY patches/Makefile_module.builtin.patch /src/Makefile_module.builtin.patch
 RUN patch < Makefile_module.builtin.patch
 RUN make -j $(($(nproc) / 2)) modules
+# All of this nonsense is to work around software that wasnt designed for static kernels
```
Comment (smira) [inline Dockerfile:20]: :rofl:

**Review by smira — APPROVED** [review]

**Comment by andrewrynhard** [comment]
I would really like to avoid any crazy workarounds as part of our official solution. IMHO I’d rather document this as a limitation and point to poorly implemented code upstream. I know we need this internally for our CI but I’m curious if we can handle it operationally using userdata  that creates empty files.

Hmm maybe we can’t. IIRC files are limited to be created relative to /var. 

I’m still hesitant to take this route. We’re not guaranteed that this will work. In fact it may have the reverse affect we would like in some cases. What if a program attempts to read something from one of these files and thinks a module doesn’t exist because the file is empty and doesn’t list anything in it. At least one of the files is expected to be binary as well. What will that break?

You see the concern I have. I’m just really hesitant to add a hack in as our official solution. Thoughts?

**Comment by smira** [comment]
what was the problem PR tries to fix? some programs depend on kernel modules while we have it all static?

**Comment by bradbeam** [comment]
> what was the problem PR tries to fix? some programs depend on kernel modules while we have it all static?

Exactly. We've hit this issue in a few different ways. Initially it was with rook where they shifted from using `modprobe` to `modules.builtin`. We were able to work around that with this patch [1] to force the kernel build to run through the modules work even though they are disabled. This got us `modules.builtin` and allowed us to proceed with the rook usage. However, we've had some instabilities as of late, so we went with an external ceph node for the time being. Unfortunately, the ceph/rbd code path[2] makes a call to modprobe to check for the rbd module. 
In testing, it seems harmless to have these empty files in place especially since there is no mechanism to load the modules and allows us to function. But 100% agree, this is a hacky workaround. After our convo earlier, it seems like the most appropriate next step is to take this up to the linux mailing list(?) and see if we can get any traction there. Something like allow the modprobe facility to support querying static kernels.

[1]
https://github.com/talos-systems/kernel/blob/master/patches/Makefile_module.builtin.patch
[2]
https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/rbd/rbd_util.go#L429-L432

**Comment by smira** [comment]
I wonder if we could have patched `modprobe` to return "success" for the fixed set of modules we compile in?

**Pull request merged**